cpp/src/io/parquet/writer_impl.cu

```diff
@@ -752,6 +758,8 @@ void writer::impl::write_chunk(table_view const &table, pq_chunked_state &state)
                                  state.stream);
   }
 
+  // Early exit for empty table
+  if (num_rows == 0 || num_columns == 0) { return; }
```

Comment:
If we're certain that we write nothing to sink in case of empty table, then this should probably be sooner. Before construction of `parquet_column` maybe. It'll avoid the `column_device_view::create()` in `get_dremel_data()`.